File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:

        n: int = len(nums)

        if n == 0:
            return 0

        maxSum: int = nums[0]
        path: int = nums[0]

        for i in range(1, n):
            path = max(nums[i], path + nums[i])
            maxSum = max(maxSum, path)

        return maxSum
        

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
#include <iostream>
#include <algorithm>

using std::max;
class Solution {
public:
    int maxSubArray(vector<int>& nums) {

        int n = nums.size();

        if(n==0){
            return 0;
        }

        int path = nums[0];
        int maxSum = nums[0];

        for(int i=1; i<n; i++){
            path = max(path+nums[i], nums[i]);
            maxSum = max(maxSum, path);
        }
        
        return maxSum;
    }
};